flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -712,7 +718,10 @@ protected void afterInvoke() throws Exception {
         // tasks. During this process, this task could coordinate with its downstream tasks to
         // continue perform checkpoints.
         CompletableFuture<Void> allRecordsProcessedFuture;
-        if (configuration.isCheckpointingEnabled()) {
+        if (configuration
```

Comment:
Below there is also a part that 

```java
// No new checkpoints could be triggered since mailbox has been drained.
subtaskCheckpointCoordinator.waitForPendingCheckpoints();
LOG.debug("All pending checkpoints are finished");
```

should also be enabled only when the flag is true and checkpoint is enabled.